src/Aspire.Hosting/Redis/RedisResource.cs

```diff
@@ -9,6 +9,13 @@ namespace Aspire.Hosting.ApplicationModel;
 /// <param name="name">The name of the resource.</param>
 public class RedisResource(string name) : ContainerResource(name), IResourceWithConnectionString
 {
+    internal const string PrimaryEndpointName = "tcp";
+
+    /// <summary>
+    /// Gets the primary endpoint for the Redis server.
+    /// </summary>
+    public EndpointReference PrimaryEndpoint => new(this, PrimaryEndpointName);
```

Comment:
Using specifically named endpoint is how I suggested we fix https://github.com/dotnet/aspire/issues/844. If we like this pattern I'll do it everywhere.